marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -236,6 +244,14 @@ lintContract env (Term t@(Pay acc payee token payment cont) pos) = do
     _ -> do
       pure unit
   markSimplification constToVal SimplifiableValue payment sa
+  case t of
+    (Pay (AccountId accNo (Term party _)) _ _ _ _) -> do
+      deposits <- use _deposits
+      let
+        positionlessAccountId = AccountId accNo (mkDefaultTerm party)
```

Comment:
I think we should use Marlowe semantics `AccountId` for this, and then it won't need the `mkDefaultTerm` I guess

Reply:
the issue there is that we have to then convert between one type of `AccountId` and another, which is fine but doesn't reduce boilerplate and introduces a `Maybe` so not sure if that is better or not, what do you think?
```haskell
class FromTerm a b where
  fromTerm :: a -> Maybe b
```

Reply:
I think that is essentially what you are doing with the `case`, right?. In fact, you are matching the whole `Pay` construct, but what you meant to match is payee, the rest will always match, I think

Reply:
yeah, I'm doing it with pattern matching on the account rather than on a maybe.

Reply:
ok, pushed change to do this, I think you're right, it's a bit clearer what is going on